dsa/src/verifying_key.rs

```diff
@@ -56,23 +65,47 @@ impl VerifyingKey {
         let (r, s) = (signature.r(), signature.s());
         let y = self.y();
 
-        if signature.r() >= q || signature.s() >= q {
+        if r >= q || s >= q {
             return Some(false);
         }
 
-        let w = s.mod_inverse(q)?.to_biguint().unwrap();
+        let q = q.widen(p.bits_precision());
+        let q = &q;
+        let r = r.widen(p.bits_precision());
+        let r = &r;
+        let s = s.widen(p.bits_precision());
+        let s = &s;
```

Comment:
This is just to save writing `&q`, `&r`, and `&s` I guess?

Reply:
Just me when I was fiddling with precisions.